src/Plugin/search_api/processor/TypedRelationFiltered.php

```diff
@@ -0,0 +1,169 @@
+    $this->entityTypeManager = $entityTypeManager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity_type.manager'),
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getPropertyDefinitions(DatasourceInterface $datasource = NULL): array {
+    $properties = [];
+
+    if (!$datasource || !$datasource->getEntityTypeId()) {
+      return $properties;
+    }
+
+    $entity_type = $datasource->getEntityTypeId();
+
+    // Get all configured typed relation fields.
+    $fields = $this->entityTypeManager->getStorage('field_config')->loadByProperties([
+      'entity_type' => $entity_type,
+      'field_type' => 'typed_relation',
+    ]);
+
+    foreach ($fields as $field) {
+      // Create a "filtered" option.
+      $definition = [
+        'label' => $this->t('@label (filtered by type) [@bundle]', [
+          '@label' => $field->label(),
+          '@bundle' => $field->getTargetBundle(),
+        ]),
+        'description' => $this->t('Typed relation field, filtered by type'),
+        'type' => 'string',
+        'processor_id' => $this->getPluginId(),
+        'is_list' => TRUE,
+        'settings' => [
+          'options' => $field->getSetting('rel_types'),
+          'bundle' => $field->getTargetBundle(),
+          'base_field' => $field->getName(),
+        ],
+      ];
+      $fieldname = 'typed_relation_filter__' . str_replace('.', '__', $field->id());
+      $property = new TypedRelationFilteredProperty($definition);
+      $property->setSetting('options', $field->getSetting('rel_types'));
+      $properties[$fieldname] = $property;
+    }
+    return $properties;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function addFieldValues(ItemInterface $item) {
+    // Skip if no Typed Relation Filtered search_api_fields are configured.
+    $relevant_search_api_fields = [];
+    $search_api_fields = $item->getFields(FALSE);
+    foreach ($search_api_fields as $search_api_field) {
+      if (substr($search_api_field->getPropertyPath(), 0, 23) == 'typed_relation_filter__') {
+        $relevant_search_api_fields[] = $search_api_field;
+      }
+    }
+    if (empty($search_api_fields)) {
+      return;
+    }
+    // Cycle over any typed relation fields on the original item.
+    $content_entity = $item->getOriginalObject()->getValue();
+
+    foreach ($relevant_search_api_fields as $search_api_field) {
+      $field_config = $search_api_field->getConfiguration();
+
+      // Exit if we're on the wrong bundle or the field isn't set.
+      if (($content_entity->bundle() != $field_config['bundle'])
+        || !$content_entity->hasField($field_config['base_field'])) {
+        return;
+      }
+
+      $vals = $content_entity->$field_config['base_field'];
```

Comment:
This is always empty in its current form.

Should it be 
```suggestion
      $vals = $content_entity->get($field_config['base_field'])->getValue();
```

Reply:
Thanks! I have tested, and it is now working for me (with @simonhm's correction - whoops!)